internal/cloudrunci/cloudrunci.go

```diff
@@ -154,16 +154,10 @@ func (s *Service) Request(method string, path string, opts ...func(*RetryOptions
 	var lastSeen error
 	resp := &http.Response{}
 	for i := 0; i < options.MaxAttempts; i++ {
-		req, err := s.NewRequest(method, path)
-		if err != nil {
-			lastSeen = err
-			continue
-		}
 		defaultClient := &http.Client{}
 
-		resp, err = defaultClient.Do(req)
-		if err != nil {
-			lastSeen = err
+		resp, lastSeen = defaultClient.Do(req)
+		if lastSeen != nil {
```

Comment:
suggestion: I realize this retry code was already in place, but this is the first I recall seeing it and we're talking about rolling it out to more use. I find with retries it's often really valuable to have some logging:

1. What's the action
2. What the current attempt we're on
3. The total number of attempts we might accept.

That might look like:

```
HTTP Request [attempt #2/5] POST http://example.com
```

Why is this valuable?

1. It shows you what to expect if you are watching the logs in real time
2. It allows familiarity with the logs to surface that you need a longer delay before making the first request, or that you might benefit from increasing the number of retries, without as much time staring at the logs

Reply:
If we had a leveled logger, i'd include this as a Debug log, but I'm doubtful that the volume of logging is useful in the vast majority of cases. (happy to revisit this when `log/slog` is an option).
I have wrapped the returned error to indicate retries, so that calling code will always know how many retries were done.